src/base_model.jl

```diff
@@ -380,6 +557,38 @@ function market_profile_by_user(ECModel::AbstractEC, u_name, profile_name)
     return profile(ECModel.market_data[user_tariff_name], profile_name)
 end
 
+"""
+    get_dhn_data(ECModel::AbstractEC, u_name)
+
+Function to retrieve district heating network (DHN) data for a user.
+If the user is connected to the DHN, it returns both the DHN data and the corresponding heat exchanger data.
+If the user is not connected, it returns the user's local heating components.
+
+## Arguments
+
+* `ECModel`: EC model object
+* `u_name`: user name
+
+## Returns
+
+It returns a dictionary containing DHN data and heat exchanger data if the user is connected.
+Otherwise, it returns the user's local heating components.
+"""
+function get_dhn_data(ECModel::AbstractEC, u_name)
+    user_data = ECModel.users_data[u_name]
+    
+    if haskey(user_data, "DHN") && user_data["DHN"]["connected"]
+        return Dict(
+            "DHN_data" => user_data["DHN"],  # District heating network data, if present
+            "ECU_data" => get(user_data, "ECU", Dict())  # Heat exchanger data
+        )
+    else
+        return Dict(
+            "local_heating_assets" => filter(p -> p.second["type"] == "thermal" || p.first == "tes", user_data)
+        )  # Return local heating assets if not connected to DHN
+    end
+end

```

Comment:
Mmmm the DHN may be a possible advancement but may be good to focus on a single user first, then the cooling. What do you think?